### PR TITLE
Doischer/3055 add report by type evaluator

### PR DESCRIPTION
**Pull request by tomdoischer**
https://redmine.czechidm.com/issues/3055

### CODE REVIEW

File: Realization/backend/rpt/rpt-impl/src/main/java/eu/bcvsolutions/idm/rpt/security/evaluator/ReportByReportTypeEvaluator.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+		}
+
+		return permissions;
+	}
+	
+	@Override
+	public List<IdmFormAttributeDto> getFormAttributes() {
+		IdmFormAttributeDto reportTypeFormAttribute = new IdmFormAttributeDto(PARAMETER_REPORT_TYPE, 
+				PARAMETER_REPORT_TYPE, 
+				PersistentType.TEXT,
```
Comment (kucera-roman) [inline Realization/backend/rpt/rpt-impl/src/main/java/eu/bcvsolutions/idm/rpt/security/evaluator/ReportByReportTypeEvaluator.java:86]: Make the param shortext, and don't use textarea face type. Now the param is rendered as big input, which is used for multi value text options in other places in IdM, but this attribute is single value

File: Realization/frontend/czechidm-rpt/src/locales/cs.json [hunk 2]
```diff
@@ -124,6 +124,17 @@
...
         }
       }
+    },
+    "authorization-evaluator": {
+      "rpt-report-by-report-type-evaluator": {
+        "help": "Uděluje přístup pouze k jednomu kódem specifikovanému reportu",
+        "attributes": {
+          "report-type": {
+            "label": "Kód reportu",
+            "help": "Kód executoru, např. 'identity-report'. Ke stažení kteréhokoli vygenerovaného reportu tohoto typu je třeba mít i oprávnění Vytvoření či Administrace!"
```
Comment (kucera-roman) [inline Realization/frontend/czechidm-rpt/src/locales/cs.json:134]: Add info that this field is for single value

File: Realization/frontend/czechidm-rpt/src/locales/en.json [hunk 3]
```diff
@@ -124,6 +124,17 @@
...
         }
       }
+    },
+    "authorization-evaluator": {
+      "rpt-report-by-report-type-evaluator": {
+        "help": "Gives access to only one report specified by code",
+        "attributes": {
+          "report-type": {
+            "label": "Report code",
+            "help": "Executor code, e. g., 'identity-report'. To allow downloading every generated report of this type, a Create or Administration permission is needed alongside Read!"
```
Comment (kucera-roman) [inline Realization/frontend/czechidm-rpt/src/locales/en.json:134]: Add info that this field is for single value

**Review by kucera-roman — COMMENTED** [review]
Look at the comments please, it's only minor fixes to improve the ux for user. Otherwise LGTM and works nicely. Thx.

File: Realization/frontend/czechidm-rpt/src/locales/cs.json [hunk 4]
```diff
@@ -131,7 +131,7 @@
         "attributes": {
           "report-type": {
             "label": "Kód reportu",
-            "help": "Kód executoru, např. 'identity-report'. Ke stažení kteréhokoli vygenerovaného reportu tohoto typu je třeba mít i oprávnění Vytvoření či Administrace!"
+            "help": "Kód executoru, např. 'identity-report'. Ke stažení kteréhokoli vygenerovaného reportu tohoto typu je třeba mít i oprávnění Vytvoření či Administrace! Toho pole je jednohodnotové, tj. může obsahovat pouze jeden report."
```
Comment (kucera-roman) [inline Realization/frontend/czechidm-rpt/src/locales/cs.json:134]: There is typo in "Toho" should be "Toto"

Comment (tomdoischer) [reply Realization/frontend/czechidm-rpt/src/locales/cs.json]: Thank you, fixed.

**Pull request merged**